approach.md

```diff
@@ -0,0 +1,60 @@
+# OSEP proposed approach
+
+a) Document scope of analysis using STPA
+
+*In collaboration with other WGs e.g. Automotive?*
+
+* Assumed system context
+    - Hardware features, role of OS, boundaries of analysis, etc
+    - Doesn't have to be concrete / complex
+      - Specific to topic: start simple and elaborate later!
+* Losses
+    - OS-related outcomes that *may* violate a system's safety goals
+      - i.e. lead to harm in a safety-related system
+* Hazards
+    - OS-level system conditions that *may* lead to these losses
+* System-level constraints
+    - Criteria that must be satisfied to *prevent* or *mitigate* hazards
+      - May be a simple inversion of the hazard
+
+b) Identify and document system measures and mitigations for Linux
+
+*Out of scope for OSEP - LFSCS group?*
+
+* Provided by kernel features, compiler, hardware, etc
+* Find supporting evidence (design, tests, processes) for these
+* Identify responsibilities of components (kernel, compiler, etc)
+
+c) Perform hazard and risk analysis using STPA
+
+ *In collaboration with LFSCS and Safety Architecture WGs?*
+
+* Based on inputs from (a) and (b)
+* Document control structure(s) for topic
+    - Identify controllers and responsibilities
+      - Functional abstraction of system elements
+      - e.g. Kernel or subsystems, complier, other tools, etc
+    - Define interactions as control actions and feedback
+    - Identify Unsafe Control Actions and Loss Scenarios
+    - Define Controller Constraints
+
```

Comment:
From Gab: At this stage I think that we would have some controllers with unsafe control actions for which we do not have architectural mitigation; for these we need to

Reply:
Breaking a controller (or a controlled process) down into sub-components, or performing a new analysis at a more detailed level of abstraction, should certainly be an option. What I'd like to discuss in the WG at some point is when we need to do this. Your suggested trigger (we can't identify a mitigation) is only one example, in my opinion; another might be that we have identified a causal factor for a UCA during Loss Scenario definition, which reveals controllers or controlled processes that are involved at a lower level of granularity.